test/lib/pgBackRestTest/Common/RunTest.pm

```diff
@@ -341,6 +344,59 @@ sub testResult
             $bDone = true;
         }
     } while (!$bDone);
+
+    # If we get here then test any warning message
+    if (defined($strWarnMessage))
+    {
+        testWarning($fnSub, $strWarnMessage, $bTestExactWarnMessage);
+    }
```

Comment:
I don't think it is safe to make two calls to `$fnSub` during a test.  If the function is stateful it may not behave the same twice in a row.  Also, this make performance worse.

The logic in test warning should be embedded in this function (or abstracted to multiple functions) so that fnSub is only called once.  Of course, only if `defined($strWarnMessage)`.